src/dsl.ts

```diff
@@ -550,76 +593,51 @@ const parensExpr = lazy(() =>
 );
 
 function BINARY_LEFT(
-  operatorsParser: Parser<string>,
-  nextParser: Parser<ParsedBinaryExpression>,
+  operatorsParser: Parser<OperatorType>,
+  nextParser: Parser<ParsedExpression>,
```

Comment:
I had these types wrong previously - correcting them was a part of not needing any of the type annotations below this.